<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Fine-mapping example • susieR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping example">
<meta property="og:description" content="susieR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">susieR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.12.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stephenslab/susieR" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Fine-mapping example</h1>
                        <h4 data-toc-skip class="author">Gao Wang</h4>
            
            <h4 data-toc-skip class="date">2022-06-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stephenslab/susieR/blob/HEAD/vignettes/finemapping.Rmd" class="external-link"><code>vignettes/finemapping.Rmd</code></a></small>
      <div class="hidden name"><code>finemapping.Rmd</code></div>

    </div>

    
    
<p>This vignette demonstrates <code>susieR</code> in the context of
genetic fine-mapping. We use simulated data of expression level of a
gene (<span class="math inline">\(y\)</span>) in <span class="math inline">\(N \approx 600\)</span> individuals. We want to
identify with the genotype matrix <span class="math inline">\(X_{N\times
P}\)</span> (<span class="math inline">\(P=1001\)</span>) the genetic
variables that causes changes in expression level.</p>
<p>The simulated data set is simulated to have exactly 3 non-zero
effects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stephenslab/susieR" class="external-link">susieR</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="the-data-set">The data-set<a class="anchor" aria-label="anchor" href="#the-data-set"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/attach.html" class="external-link">attach</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">)</span></code></pre></div>
<p>The loaded dataset contains regression data <span class="math inline">\(X\)</span> and <span class="math inline">\(y\)</span>, along with some other relevant
properties in the context of genetic studies. It also contains the
“true” regression coefficent the data is simulated from.</p>
<p>Notice that we’ve simulated 2 sets of <span class="math inline">\(Y\)</span> as 2 simulation replicates. Here we’ll
focus on the first data-set.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>
<span class="co"># [1] 574   2</span></code></pre></div>
<p>Here are the 3 “true” signals in the first data-set:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">b</span> <span class="op">&lt;-</span> <span class="va">true_coef</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">b</span>, pch<span class="op">=</span><span class="fl">16</span>, ylab<span class="op">=</span><span class="st">'effect size'</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-4-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">b</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span>
<span class="co"># [1] 403 653 773</span></code></pre></div>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div class="section level2">
<h2 id="simple-regression-summary-statistics">Simple regression summary statistics<a class="anchor" aria-label="anchor" href="#simple-regression-summary-statistics"></a>
</h2>
<p><code>univariate_regression</code> function can be used to compute
summary statistics by fitting univariate simple regression variable by
variable. The results are <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(SE(\hat{\beta})\)</span> from which z-scores can
be derived. Again we focus only on results from the first data-set:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/univariate_regression.html">univariate_regression</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">z_scores</span> <span class="op">&lt;-</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">betahat</span> <span class="op">/</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">sebetahat</span>
<span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">z_scores</span>, y <span class="op">=</span> <span class="st">"z"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-6-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
<div class="section level2">
<h2 id="fine-mapping-with-susier">Fine-mapping with <code>susieR</code><a class="anchor" aria-label="anchor" href="#fine-mapping-with-susier"></a>
</h2>
<p>For starters, we assume there are at most 10 causal variables, i.e.,
set <code>L = 10</code>, although SuSiE is robust to the choice of
<code>L</code>.</p>
<p>The <code>susieR</code> function call is:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fitted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/susie.html">susie</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,
                L <span class="op">=</span> <span class="fl">10</span>,
        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># [1] "objective:-1380.57545244487"</span>
<span class="co"># [1] "objective:-1377.4866091747"</span>
<span class="co"># [1] "objective:-1375.85777210115"</span>
<span class="co"># [1] "objective:-1375.80892303931"</span>
<span class="co"># [1] "objective:-1370.33949333171"</span>
<span class="co"># [1] "objective:-1370.19677276994"</span>
<span class="co"># [1] "objective:-1370.10919739202"</span>
<span class="co"># [1] "objective:-1370.10918017469"</span>
<span class="co"># [1] "objective:-1370.10901872278"</span></code></pre></div>
<div class="section level3">
<h3 id="credible-sets">Credible sets<a class="anchor" aria-label="anchor" href="#credible-sets"></a>
</h3>
<p>By default, we output 95% credible set:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">$</span><span class="va">sets</span><span class="op">)</span>
<span class="co"># $cs</span>
<span class="co"># $cs$L2</span>
<span class="co"># [1] 653</span>
<span class="co"># </span>
<span class="co"># $cs$L1</span>
<span class="co"># [1] 773 777</span>
<span class="co"># </span>
<span class="co"># $cs$L3</span>
<span class="co">#  [1] 362 365 372 373 374 379 381 383 384 386 387 388 389 391 392 396 397 398 399</span>
<span class="co"># [20] 400 401 403 404 405 407 408 415</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># $purity</span>
<span class="co">#    min.abs.corr mean.abs.corr median.abs.corr</span>
<span class="co"># L2    1.0000000     1.0000000       1.0000000</span>
<span class="co"># L1    0.9815726     0.9815726       0.9815726</span>
<span class="co"># L3    0.8686309     0.9640176       0.9720711</span>
<span class="co"># </span>
<span class="co"># $cs_index</span>
<span class="co"># [1] 2 1 3</span>
<span class="co"># </span>
<span class="co"># $coverage</span>
<span class="co"># [1] 0.9998236 0.9988858 0.9539811</span>
<span class="co"># </span>
<span class="co"># $requested_coverage</span>
<span class="co"># [1] 0.95</span></code></pre></div>
<p>The 3 causal signals have been captured by the 3 CS reported here.
The 3rd CS contains many variables, including the true causal variable
<code>403</code>. The minimum absolute correlation is 0.86.</p>
<p>If we use the default 90% coverage for credible sets, we still
capture the 3 signals, but “purity” of the 3rd CS is now 0.91 and size
of the CS is also a bit smaller.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/susie_get_methods.html">susie_get_cs</a></span><span class="op">(</span><span class="va">fitted</span>,
                     X <span class="op">=</span> <span class="va">X</span>,
             coverage <span class="op">=</span> <span class="fl">0.9</span>,
                     min_abs_corr <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">sets</span><span class="op">)</span>
<span class="co"># $cs</span>
<span class="co"># $cs$L2</span>
<span class="co"># [1] 653</span>
<span class="co"># </span>
<span class="co"># $cs$L1</span>
<span class="co"># [1] 773 777</span>
<span class="co"># </span>
<span class="co"># $cs$L3</span>
<span class="co">#  [1] 373 374 379 381 383 384 386 387 388 389 391 392 396 398 399 400 401 403 404</span>
<span class="co"># [20] 405 407 408</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># $purity</span>
<span class="co">#    min.abs.corr mean.abs.corr median.abs.corr</span>
<span class="co"># L2    1.0000000     1.0000000       1.0000000</span>
<span class="co"># L1    0.9815726     0.9815726       0.9815726</span>
<span class="co"># L3    0.9119572     0.9726283       0.9765888</span>
<span class="co"># </span>
<span class="co"># $cs_index</span>
<span class="co"># [1] 2 1 3</span>
<span class="co"># </span>
<span class="co"># $coverage</span>
<span class="co"># [1] 0.9998236 0.9988858 0.9119917</span>
<span class="co"># </span>
<span class="co"># $requested_coverage</span>
<span class="co"># [1] 0.9</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="posterior-inclusion-probabilities">Posterior inclusion probabilities<a class="anchor" aria-label="anchor" href="#posterior-inclusion-probabilities"></a>
</h3>
<p>Previously we’ve determined that summing over 3 single effect
regression models is approperate for our application. Here we summarize
the variable selection results by posterior inclusion probability
(PIP):</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fitted</span>, y<span class="op">=</span><span class="st">"PIP"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-11-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>The true causal variables are colored red. The 95% CS identified are
circled in different colors. Of interest is the cluster around position
400. The true signal is 403 but apparently it does not have the highest
PIP. To compare ranking of PIP and original z-score in that CS:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">i</span>  <span class="op">&lt;-</span> <span class="va">fitted</span><span class="op">$</span><span class="va">sets</span><span class="op">$</span><span class="va">cs</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>
<span class="va">z3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">i</span>,<span class="va">z_scores</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,<span class="va">fitted</span><span class="op">$</span><span class="va">pip</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">z3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'position'</span>, <span class="st">'z-score'</span>, <span class="st">'PIP'</span><span class="op">)</span>
<span class="va">z3</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">z3</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="co">#       position  z-score         PIP</span>
<span class="co">#  [1,]      396 5.189811 0.056704331</span>
<span class="co">#  [2,]      381 5.164794 0.100360243</span>
<span class="co">#  [3,]      386 5.164794 0.100360243</span>
<span class="co">#  [4,]      379 5.077563 0.054179507</span>
<span class="co">#  [5,]      391 5.068388 0.055952118</span>
<span class="co">#  [6,]      383 5.057053 0.052896918</span>
<span class="co">#  [7,]      384 5.057053 0.052896918</span>
<span class="co">#  [8,]      389 5.052519 0.042161265</span>
<span class="co">#  [9,]      405 5.039617 0.045761975</span>
<span class="co"># [10,]      403 5.035949 0.031992848</span>
<span class="co"># [11,]      387 5.013526 0.041041505</span>
<span class="co"># [12,]      388 4.997955 0.039650079</span>
<span class="co"># [13,]      408 4.994865 0.041551961</span>
<span class="co"># [14,]      404 4.954407 0.032013339</span>
<span class="co"># [15,]      374 4.948060 0.030571484</span>
<span class="co"># [16,]      373 4.934410 0.023577221</span>
<span class="co"># [17,]      362 4.894243 0.012145481</span>
<span class="co"># [18,]      399 4.860780 0.026454056</span>
<span class="co"># [19,]      392 4.856384 0.019741011</span>
<span class="co"># [20,]      407 4.849285 0.014699313</span>
<span class="co"># [21,]      400 4.827361 0.021659443</span>
<span class="co"># [22,]      365 4.782770 0.006263425</span>
<span class="co"># [23,]      398 4.751205 0.012907848</span>
<span class="co"># [24,]      401 4.723184 0.014858460</span>
<span class="co"># [25,]      397 4.716886 0.008690915</span>
<span class="co"># [26,]      415 4.663208 0.009003129</span>
<span class="co"># [27,]      372 4.581560 0.005886458</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="choice-of-priors">Choice of priors<a class="anchor" aria-label="anchor" href="#choice-of-priors"></a>
</h3>
<p>Notice that by default SuSiE estimates prior effect size from data.
For fine-mapping applications, however, we sometimes have knowledge of
SuSiE prior effect size since it is parameterized as percentage of
variance explained (PVE) by a non-zero effect, which, in the context of
fine-mapping, is related to per-SNP heritability. It is possible to use
<code>scaled_prior_variance</code> to specify this PVE and explicitly
set <code>estimate_prior_variance=FALSE</code> to fix the prior effect
to given value.</p>
<p>In this data-set, SuSiE is robust to choice of priors. Here we set
PVE to 0.2, and compare with previous results:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fitted</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie.html">susie</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,
               L <span class="op">=</span> <span class="fl">10</span>,
               estimate_residual_variance <span class="op">=</span> <span class="cn">TRUE</span>, 
               estimate_prior_variance <span class="op">=</span> <span class="cn">FALSE</span>, 
               scaled_prior_variance <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>
<span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fitted</span>, y<span class="op">=</span><span class="st">'PIP'</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-13-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>which largely remains unchanged.</p>
</div>
<div class="section level3">
<h3 id="a-note-on-covariate-adjustment">A note on covariate adjustment<a class="anchor" aria-label="anchor" href="#a-note-on-covariate-adjustment"></a>
</h3>
<p>To include covariate <code>Z</code> in SuSiE, one approach is to
regress it out from both <code>y</code> and <code>X</code>, and then run
SuSiE on the residuals. The code below illustrates the procedure:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>remove.covariate.effects <span class="ot">&lt;-</span> <span class="cf">function</span> (X, Z, y) {</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># include the intercept term</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">any</span>(Z[,<span class="dv">1</span>]<span class="sc">!=</span><span class="dv">1</span>)) Z <span class="ot">=</span> <span class="fu">cbind</span>(<span class="dv">1</span>, Z)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  A   <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">crossprod</span>(Z))</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  SZy <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">solve</span>(A,<span class="fu">c</span>(y <span class="sc">%*%</span> Z)))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  SZX <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">solve</span>(A,<span class="fu">t</span>(Z) <span class="sc">%*%</span> X))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> y <span class="sc">-</span> <span class="fu">c</span>(Z <span class="sc">%*%</span> SZy)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> X <span class="sc">-</span> Z <span class="sc">%*%</span> SZX</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">X =</span> X,<span class="at">y =</span> y,<span class="at">SZy =</span> SZy,<span class="at">SZX =</span> SZX))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>out <span class="ot">=</span> <span class="fu">remove.covariate.effects</span>(X, Z, Y[,<span class="dv">1</span>])</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>fitted_adjusted <span class="ot">=</span> <span class="fu">susie</span>(out<span class="sc">$</span>X, out<span class="sc">$</span>y,</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">L =</span> <span class="dv">10</span>) ,</span></code></pre></div>
<p>Note that the covariates <code>Z</code> should have a column of ones
as the first column. If not, the above function
<code>remove.covariate.effects</code> will add such a column to
<code>Z</code> before regressing it out. Data will be centered as a
result. Also the scale of data is changed after regressing out
<code>Z</code>. This introduces some subtleties in terms of interpreting
the results. For this reason, we provide covariate adjustment procedure
as a tip in the documentation and not part of
<code><a href="../reference/susie.html">susieR::susie()</a></code> function. Cautions should be taken when
applying this procedure and interpreting the result from it.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># R version 3.6.2 (2019-12-12)</span>
<span class="co"># Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co"># Running under: macOS Catalina 10.15.7</span>
<span class="co"># </span>
<span class="co"># Matrix products: default</span>
<span class="co"># BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span>
<span class="co"># LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span>
<span class="co"># </span>
<span class="co"># locale:</span>
<span class="co"># [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co"># </span>
<span class="co"># attached base packages:</span>
<span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co"># </span>
<span class="co"># other attached packages:</span>
<span class="co"># [1] susieR_0.12.10</span>
<span class="co"># </span>
<span class="co"># loaded via a namespace (and not attached):</span>
<span class="co">#  [1] tidyselect_1.1.1   xfun_0.29          bslib_0.3.1        purrr_0.3.4       </span>
<span class="co">#  [5] lattice_0.20-38    colorspace_1.4-1   vctrs_0.3.8        generics_0.0.2    </span>
<span class="co">#  [9] htmltools_0.5.2    yaml_2.2.0         utf8_1.1.4         rlang_0.4.11      </span>
<span class="co"># [13] mixsqp_0.3-46      pkgdown_2.0.2      jquerylib_0.1.4    pillar_1.6.2      </span>
<span class="co"># [17] glue_1.4.2         DBI_1.1.0          RcppZiggurat_0.1.5 matrixStats_0.61.0</span>
<span class="co"># [21] lifecycle_1.0.0    plyr_1.8.5         stringr_1.4.0      munsell_0.5.0     </span>
<span class="co"># [25] gtable_0.3.0       ragg_0.3.1         memoise_1.1.0      evaluate_0.14     </span>
<span class="co"># [29] knitr_1.37         fastmap_1.1.0      parallel_3.6.2     irlba_2.3.3       </span>
<span class="co"># [33] fansi_0.4.0        Rfast_2.0.3        highr_0.8          Rcpp_1.0.7        </span>
<span class="co"># [37] scales_1.1.0       backports_1.1.5    desc_1.2.0         jsonlite_1.7.2    </span>
<span class="co"># [41] systemfonts_1.0.2  fs_1.5.2           ggplot2_3.3.5      digest_0.6.23     </span>
<span class="co"># [45] stringi_1.4.3      dplyr_1.0.7        grid_3.6.2         rprojroot_1.3-2   </span>
<span class="co"># [49] tools_3.6.2        magrittr_2.0.1     sass_0.4.0         tibble_3.1.3      </span>
<span class="co"># [53] crayon_1.4.1       pkgconfig_2.0.3    ellipsis_0.3.2     Matrix_1.2-18     </span>
<span class="co"># [57] assertthat_0.2.1   rmarkdown_2.11     reshape_0.8.8      R6_2.4.1          </span>
<span class="co"># [61] compiler_3.6.2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Gao Wang, Yuxin Zou, Kaiqian Zhang, Peter Carbonetto,
Matthew Stephens.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
